contracts/astroport-liquidity-helper/src/contract.rs

```diff
@@ -190,93 +186,51 @@ pub fn execute_balancing_provide_liquidity(
-            // For stable pools we are allowed to provide liquidity in any ratio,
-            // so we simply provide liquidity with all passed assets.
-            let provide_liquidity_res =
-                pool.provide_liquidity(deps.as_ref(), &env, assets.clone(), min_out)?;
-
-            // Callback to return LP tokens
-            let callback_msg = CallbackMsg::ReturnLpTokens {
-                pool,
-                balance_before: lp_token_balance,
-                recipient,
-            }
-            .into_cosmos_msg(&env)?;
-
-            let event =
-                Event::new("apollo/astroport-liquidity-helper/execute_balancing_provide_liquidity")
-                    .add_attribute("action", "stable_provide_liquidity")
-                    .add_attribute("assets", assets.to_string())
-                    .add_attribute("min_out", min_out);
-
-            Ok(merge_responses(vec![receive_res, provide_liquidity_res])
-                .add_message(callback_msg)
-                .add_event(event))
-        }
-        PairType::Custom(_) => Err(ContractError::CustomPairType {}),
+        PairType::Custom(t) => match t.as_str() {
+            "concentrated" => Response::new(),
+            _ => return Err(ContractError::UnsupportedPairType {}),
+        },
+        _ => Response::new(),
+    };
+
+    // For stableswap and concentrated liquidity pools we are allowed to provide
+    // liquidity in any ratio, so we simply provide liquidity with all passed
+    // assets.
+    let provide_liquidity_res =
+        pool.provide_liquidity(deps.as_ref(), &env, assets.clone(), min_out)?;
+
+    // Callback to return LP tokens
+    let callback_msg = CallbackMsg::ReturnLpTokens {
+        pool,
+        balance_before: lp_token_balance,
+        recipient,
     }
-}
-
-/// CallbackMsg handler to provide liquidity with the given assets. This needs
-/// to be a callback, rather than doing in the first ExecuteMsg, because
-/// pool.provide_liquidity does a simulation with current reserves, and we do a
-/// swap in the top level ExecuteMsg, which means the reserves would be wrong in
-/// the provide liquidity simulation.
-pub fn execute_callback_provide_liquidity(
-    deps: DepsMut,
-    env: Env,
-    _info: MessageInfo,
-    assets: AssetList,
-    min_out: Uint128,
-    pool: AstroportPool,
-) -> Result<Response, ContractError> {
-    let res = pool.provide_liquidity(deps.as_ref(), &env, assets.clone(), min_out)?;
-
-    let event = Event::new("apollo/astroport-liquidity-helper/execute_callback_provide_liquidity")
-        .add_attribute("assets", assets.to_string());
-
-    Ok(res.add_event(event))
+    .into_cosmos_msg(&env)?;
+
+    let event: Event =
+        Event::new("apollo/astroport-liquidity-helper/execute_balancing_provide_liquidity")
+            .add_attribute("assets", assets.to_string())
+            .add_attribute("min_out", min_out);
+
+    Ok(
+        merge_responses(vec![receive_res, swap_res, provide_liquidity_res])
```

Comment:
Just so I understand this correctly. The provide message for all pool types is constructed with `pool.provide_liquidity` and this will use the liquidity manager contract for all pools. `receive_res` is the same for all and for `swap_res` only XYK pools need swap because PCL and Stable support single sided provision. Is that correct?

Reply:
Yeah exactly